### PR TITLE
Release on 2022-04-06

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We would like to make a new release.  The key changes are:
- Merge neuron functionality (using quill, not the GUI).
- Updated logo
- UI improvements
- More tests

Release commit: 453d27379f3c5271af011464aada554be8bc9a4d
CI Artifacts: https://github.com/dfinity/nns-dapp/actions/runs/2095628660
SHA256sum:
  - CI: 9057f2bd4a28ec2d83fb9a9f7c66290b2c4aa8454b4f093c289e8c924d8e05e2  nns-dapp.wasm
  - Max: 9057f2bd4a28ec2d83fb9a9f7c66290b2c4aa8454b4f093c289e8c924d8e05e2  nns-dapp.wasm

# Changes
- Update CHANGELOG.md

# Tests
The release candidate has been deployed to a testnet:
- https://nns-prod.winning.gold == https://qsgjb-riaaa-aaaaa-aaaga-cai.nnsdapp.dfinity.network/

### Key features
- Hamish has tested the merge neurons functionality.
- The logo is 100% on-chain.

#### Chrome + Linux
- Login + logout per https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#469c96ea4e6c45a8867cddbf97f5750d
- Transfer funds per https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#0929055c81dd4f009242ef8e1fb0d832
  - TODO: If I have two tabs open and transact in one, the other tab does not get updated.  Moving around tabs is also not enough to trigger an update.  We should consider putting the data store in local storage and listening for updates.  Or at least having a trigger that other tabs can listen for and use to update their local data.
- Sync multiple tabs for login state: https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#23f043fec1f444ae9b916ce6ce88c495 
- Skipping: Deep links are tested, insofar as they work, by the e2e redirect test.  https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#3555ab090bf742a9948b73fa382ba098
- Known neurons are shown: https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#bea7856be7ee436982241b9f56fcab28
  - Yes, but there is only the standard set of 2.  We should vote in new neurons.
  - TODO:  UI to create arbitrary proposals including create known neurons.
  - TODO: Ability to pass proposals on the testnet.  A God neuron with > 50% of stake.
- Neurons can be disbursed: https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#285d2753325d41828df9a54a1a845f4f
- Skipping NA: Neurons can be merged - quill only.
- Proposaals are shown: https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#9bfeaaedd970419c9b4de549523e0f94
- Proposal filtering works: https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#6198a0b695ff4c1c920da474e1b92eca
  - Note: Some filters returned empty results because all proposals are new, eligible, not yet disbursed etc.
- Proposer information is available  
  - https://www.notion.so/NNS-Dapp-Manual-Testing-Runbook-75ae361334324440abe8a797822b4a1b#4a5a039fc6eb4998949d7ac725155ae1
- Flutter does not have infinite scroll.
  - TODO: Infinite scroll is an implementation detail.  "All proposals are visible" is a user requirement.  Infinite scroll is a nice way of doing this but it specifies a solution, not a requirement.  A button "show more" may be clunky but should also be covered by the test.
- Users can vote
  - TODO: Proposal representation kinda sucks.  Can we show e.g. gps information on a map?
  - TODO: Given that this is the most important of all, move it close to the top of the list.

#### Linux Firefox
Login, scroll through all the tabs, logout.

#### Chrome on M1 Mac
Login, scroll through all the tabs, logout.

#### Safari on M1 Mac
Black screen.  Safari is not on the supported list, I was just curious.  Ditto Intel mac.

TODO: Landing page that says "Please install a sensible browser"?

#### FF on Intel Mac
Login, scroll through all the tabs, logout.

## e2e tests
The "home" test fails, even though logging in clearly works when tested manually.  This is presumably a problem with the test.  Re-ran.  This time it is all green.

```
e2e-tests $ REDIRECT_TO_LEGACY=prod ./scripts/test-winning
```

Screenshot from 2022-04-06 04-14-33

TODO: It would be useful to be able to filter out wdio tests based on the value of REDIRECT_TO_LEGACY

# Proposal

TODO: The proposal has e.g. git commit hashes.  A pretty version of the proposal would link each commit to github and the PR references to the corresponding PR.

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
Weird, I tried locally and this is the result:

Screenshot 2022-04-06 at 08 02 56

File: release/PROPOSAL.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+```
+
+## Wasm Verification
+
+To build the wasm module yourself and verify its hash, run the following commands from the root of the nns-dapp repo:
+
+git pull  # to ensure you have the latest changes.
+git checkout `453d27379f3c5271af011464aada554be8bc9a4d`
+./scripts/docker-build
+sha256sum nns-dapp.wasm
```
Comment (bitdivine) [inline release/PROPOSAL.md:147]: Note:  This file is NOT to be merged, it is included here for review.

File: release/PROPOSAL.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# Upgrade frontend NNS Dapp canister to commit `453d27379f3c5271af011464aada554be8bc9a4d`
+Wasm sha256 hash: `9057f2bd4a28ec2d83fb9a9f7c66290b2c4aa8454b4f093c289e8c924d8e05e2` (`https://github.com/dfinity/nns-dapp/actions/runs/2095628660`)
+
+## Change Log:
+- Merge neuron functionality (using quill, not the GUI)
+- Links in proposals fixed
+- Updated logo
+- UI improvements
+- More tests
```
Comment (peterpeterparker) [inline release/PROPOSAL.md:9]: You can maybe add "support for links in proposal detail"?

https://github.com/dfinity/nns-dapp/pull/596

Comment (peterpeterparker) [reply release/PROPOSAL.md:9]: Oh that's what you meant with "Links in proposals fixed" 😉

Comment (bitdivine) [reply release/PROPOSAL.md:9]: Your version sounds better.

**Comment by bitdivine** [comment]
Release was reverted due to a redirect loop.

**Pull request closed without merging**